pkg/httphelper/httphelper.go

```diff
@@ -0,0 +1,57 @@
+package httphelper
+
+import (
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type TraceResponseWriter struct {
+	http.ResponseWriter
+	statusCode int
+	size       int
+}
+
+func (w *TraceResponseWriter) WriteHeader(code int) {
+	w.statusCode = code
+	w.ResponseWriter.WriteHeader(code)
+}
+
+func (w *TraceResponseWriter) Write(data []byte) (int, error) {
+	size, err := w.ResponseWriter.Write(data)
+	w.size += size
+	return size, err
+}
+
+type Metrics struct {
+	HttpRequestDuration *prometheus.HistogramVec
+	HttpResponseSize    *prometheus.HistogramVec
+	ErrorRate           *prometheus.CounterVec
+}
+
+func RecordError(label string, metrics *Metrics) {
+	labels := prometheus.Labels{"error": label}
+	metrics.ErrorRate.With(labels).Inc()
+}
+
+func HandleWithMetrics(h http.HandlerFunc, metrics *Metrics) http.HandlerFunc {
```

Comment:
Also, what would be better - leaving it like this or making this function and the RecordErrors function part of the Metrics class?

Reply:
Go doesn't have classes, it only has types. And yeah, making it a method on `*Metrics` makes sense